### PR TITLE
Added ability to get headers as associative array

### PR DESCRIPTION
**Pull request by mazanax**
Implemented https://github.com/sendgrid/sendgrid-php/issues/361

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/php-http-client?pullRequest=19) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/php-http-client?pullRequest=19) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/php-http-client?pullRequest=19) it.</sub>

**Comment by thinkingserious** [comment]
Hello @mazanax,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

File: lib/Response.php [hunk 1]
```diff
@@ -64,10 +64,53 @@ public function body()
...
+    /**
+      * Returns response headers as associative array
+      * 
+      * @param array $headers
+      *
+      * @return array
+      * 
+      * @throws \InvalidArgumentException
+      */
+    private function prettifyHeaders($headers)
```
Comment (alexpts) [inline lib/Response.php]: private function prettifyHeaders(array $headers)

File: lib/Response.php [hunk 2]
```diff
@@ -64,10 +64,53 @@ public function body()
...
+    {
+        if (!is_array($headers)) {
```
Comment (alexpts) [inline lib/Response.php]: remove condition and exception, you can pass only array type for first argument

File: lib/Response.php [hunk 3]
```diff
@@ -64,10 +64,53 @@ public function body()
...
+            throw new \InvalidArgumentException('$headers should be array');
+        }
+
+        return array_reduce(
+            array_filter($headers),
```
Comment (alexpts) [inline lib/Response.php]: Remove array_filter, you can filter in reduce function

File: lib/Response.php [hunk 4]
```diff
@@ -64,10 +64,53 @@ public function body()
...
+            function ($result, $header) {
```
Comment (alexpts) [inline lib/Response.php]: function (array $result, $header) {

File: lib/Response.php [hunk 5]
```diff
@@ -64,10 +64,53 @@ public function body()
...
+                if (empty($header)) {
+                    return $result;
+                }
+
+                if (false === strpos($header, ':')) {
```
Comment (alexpts) [inline lib/Response.php]: Why header not name? Can you provider description this case?
I think is dirty  hack

**Comment by alexpts** [comment]
@mazanax, I added some minor comment for PR

File: lib/Response.php (hunk 3 above)
Comment (alexpts) [inline lib/Response.php]: You already filter it in reduce

**Comment by thinkingserious** [comment]
Hi @alexpts,

Thanks for taking the time to review!

Would you mind creating your own PR, since this one is already merged?

Thanks!

With Best Regards,

Elmer

**Pull request merged**